src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs

```diff
@@ -103,6 +113,19 @@ public void InvalidArguments_Access()
                 // Write-only access is not allowed
                 AssertExtensions.Throws<ArgumentException>("access", () => MemoryMappedFile.CreateFromFile(fs, CreateUniqueMapName(), 4096, MemoryMappedFileAccess.Write, HandleInheritability.None, true));
             }
+
+            // Test the same things, but with a SafeFileHandle
+            using (TempFile file = new TempFile(GetTestFilePath()))
+            using (FileStream fs = File.Open(file.Path, FileMode.Open))
+            {
+                SafeFileHandle fileHandle = fs.SafeFileHandle;
```

Comment:
This code is 100% correct, but we can improve it by not creating the `FileStream` and then using it's `SafeFileHandle`, but rather using the `File.OpenHandle` and work with handle directly.

The reason why I am asking for this change is that when we introduce new APIs we often don't provide docs with examples for them, so some users use our tests as examples. And if they want to use the new `SafeFileHandle` based APIs, they should not be creating `FileStream` up-front. `FileStream` is more expensive in terms of allocations (it's simply bigger and might allocate it's own buffer), moreover accessing `FileStream.SafeFileHandle` is performing a sys-call, which is expensive:

https://github.com/dotnet/runtime/blob/9eabb99670d4c0f26010451833b8f62ebbe60e32/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/OSFileStreamStrategy.cs#L96-L105

```suggestion
            using (SafeFileHandle fileHandle = File.OpenHandle(file.Path, FileMode.Open))
            {
```